tests/std/tests/P0980R1_constexpr_strings/test.cpp

```diff
@@ -0,0 +1,1758 @@
+            } else if constexpr (is_same_v<CharType, char32_t>) {
+                assert(c4 == 3);
+            } else {
+                assert(c4 == 15);
+            }
+        }
+    }
+
+    { // clear
+        str cleared = get_literal_input<CharType>();
+        cleared.clear();
+        assert(cleared.empty());
+        assert(cleared.capacity() == str(get_literal_input<CharType>()).capacity());
+    }
+
+    { // insert
+#if defined(MSVC_INTERNAL_TESTING) || defined(__EDG__) // TRANSITION, VSO-1275530
+        str insert_char               = get_literal_input<CharType>();
+        const CharType to_be_inserted = CharType{','};
+        insert_char.insert(insert_char.begin() + 5, to_be_inserted);
+        assert(equalRanges(insert_char, "Hello, fluffy kittens"sv));
+
+        str insert_const_char = get_literal_input<CharType>();
+        insert_const_char.insert(insert_const_char.cbegin() + 5, to_be_inserted);
+        assert(equalRanges(insert_const_char, "Hello, fluffy kittens"sv));
+
+        str insert_char_rvalue = get_literal_input<CharType>();
+        insert_char_rvalue.insert(insert_char_rvalue.begin() + 5, CharType{','});
+        assert(equalRanges(insert_char_rvalue, "Hello, fluffy kittens"sv));
+
+        str insert_const_char_rvalue = get_literal_input<CharType>();
+        insert_const_char_rvalue.insert(insert_const_char_rvalue.cbegin() + 5, CharType{','});
+        assert(equalRanges(insert_const_char_rvalue, "Hello, fluffy kittens"sv));
+
+        str insert_range(2, CharType{'b'});
+        const auto it = insert_range.insert(
+            insert_range.begin() + 1, begin(get_view_input<CharType>()), end(get_view_input<CharType>()));
+        assert(it == insert_range.begin() + 1);
+        assert(equalRanges(insert_range, "bHello fluffy kittensb"sv));
+
+        str insert_const_range(2, CharType{'b'});
+        const auto cit = insert_const_range.insert(
+            insert_const_range.cbegin() + 1, begin(get_view_input<CharType>()), end(get_view_input<CharType>()));
+        assert(cit == insert_const_range.cbegin() + 1);
+        assert(equalRanges(insert_const_range, "bHello fluffy kittensb"sv));
+
+        str insert_initializer = get_literal_input<CharType>();
+        const auto it_ilist    = insert_initializer.insert(insert_initializer.begin() + 6, {'c', 'u', 't', 'e', ' '});
+        assert(it_ilist == insert_initializer.begin() + 6);
+        assert(equalRanges(insert_initializer, "Hello cute fluffy kittens"sv));
+
+        str insert_const_initializer = get_literal_input<CharType>();
+        const auto cit_ilist =
+            insert_const_initializer.insert(insert_const_initializer.cbegin() + 6, {'c', 'u', 't', 'e', ' '});
+        assert(cit_ilist == insert_const_initializer.cbegin() + 6);
+        assert(equalRanges(insert_const_initializer, "Hello cute fluffy kittens"sv));
+
+        str insert_pos_str  = get_literal_input<CharType>();
+        const str to_insert = "cute and scratchy ";

```

Comment:
In MSVC-internal testing, I'm seeing a compiler error on this line:

```
.\test.cpp(577): error C2440: 'initializing': cannot convert from 'const char [19]' to 'std::basic_string<char8_t,std::char_traits<char8_t>,std::allocator<char8_t>>'
.\test.cpp(577): note: No constructor could take the source type, or constructor overload resolution was ambiguous
.\test.cpp(1711): note: see reference to function template instantiation 'bool test_interface<char8_t>(void)' being compiled
```

The problem is `char` vs. `char8_t` mismatch.

Reply:
I have a fix and will push it with my next batch of changes.